src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -38,6 +39,35 @@ void HandleRestart(Layer * systemLayer, void * appState)
 }
 } // namespace
 
+bool OTAImageProcessorImpl::IsFirstImageRun()
+{
+    OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return false;
+    }
+
+    return requestor->GetCurrentUpdateState() == OTARequestorInterface::OTAUpdateStateEnum::kApplying;
+}
+
+CHIP_ERROR OTAImageProcessorImpl::ConfirmCurrentImage()
+{
+    OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    uint32_t currentVersion;
+    ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSoftwareVersion(currentVersion));
```

Comment:
FYI, this was a simple implementation done for Darwin/Linux platforms to check the image. I'm not sure if ESP32 following that as an example is necessarily the right thing to do?

Reply:
ESP32 don’t have platform specific API's.